### PR TITLE
ENG 601 update setup to start Anvil and generate MCR config.

### PR DESCRIPTION
**Pull request by musitdev**
# Summary
- **RFCs**: [Link to RFC](#./link/to/rfc), [Link to RFC](#./link/to/rfc), or $\emptyset$.
- **Categories**: any of `protocol-units`, `networks`, `scripts`, `util`, `cicd`, or `misc`.

Start all necessary services / process to run settlement client in local.

Open for review before rebase of main branch.

# Changelog

Update Settlement Local Setup process to start Anvil and publish the MCR smart contract
Update Settlement config accordingly with anvil startup and SC deployement.
Add Suzuka Node gracefull shutdown to stop anvil process if any and call all destructor.

# Testing
Before starting Suzuka node the config.toml file in .movement must be deleted otherwise the anvil process is not started because a private key is already defined (defined by the last execution).

 To run use the Suzuka client tests :
`CELESTIA_APP_PATH=/usr/local/bin/celestia-appd CELESTIA_LOG_LEVEL=FATAL MOVEMENT_PREBUILT=false nix develop --extra-experimental-features nix-command --extra-experimental-features flakes --command bash  -c "just suzuka-full-node native test.build.setup.local --keep-tui"`

# Outstanding issues

Settlement client tests hasn't been updated and still use bash script to start anvil.

The config.toml file is not deleted at the end of execution. It will be resolved with the next config management process.

### CODE REVIEW

File: networks/suzuka/suzuka-full-node/src/main.rs
Comment (mzabaluev) [inline networks/suzuka/suzuka-full-node/src/main.rs:32]: These should not be unconditional printouts, use `tracing::debug!`?

File: networks/suzuka/suzuka-full-node/src/main.rs
Comment (mzabaluev) [inline networks/suzuka/suzuka-full-node/src/main.rs:43]: The block seems unnecessary, but this is just stylistics.

File: protocol-units/da/m1/setup/src/local.rs
Comment (mzabaluev) [inline protocol-units/da/m1/setup/src/local.rs:5]: Nitpick: I like to group use statements by dependencies in the same namespace, external crates, and `std`/`core`.

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (mzabaluev) [inline protocol-units/settlement/mcr/client/src/eth_client.rs:30]: And this is why I don't like the ungrouped imports, everything is mashed together. Ah well.

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (mzabaluev) [inline protocol-units/settlement/mcr/client/src/eth_client.rs:85]: I don't think this belongs in the client.

In the production use case, we don't run Anvil, so let's not make this a wart on `Client`, but compose this helper with `Client` somewhere in test infrastructure?

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (mzabaluev) [inline protocol-units/settlement/mcr/client/src/eth_client.rs:284]: @l-monninger did not document the separation of concerns when adding the *-setup infrastructure crates, but as I understand it, this setup functionality belongs precisely there, not in the client crate. The integration tests would then consume the configuration created by the e2e setup.

File: util/commander/src/lib.rs
Comment (mzabaluev) [inline util/commander/src/lib.rs:85]: Use `tracing`?

File: protocol-units/settlement/mcr/setup/src/local.rs
Comment (mzabaluev) [inline protocol-units/settlement/mcr/setup/src/local.rs:54]: 👍

File: protocol-units/settlement/mcr/client/src/eth/mod.rs
Comment (mzabaluev) [inline protocol-units/settlement/mcr/client/src/eth/mod.rs:3]: This is dead code and does not refer to anything yet.

File: protocol-units/settlement/mcr/config/src/lib.rs
Comment (mzabaluev) [inline protocol-units/settlement/mcr/config/src/lib.rs:28]: Can we make management of infrastructure processes part of the orchestration layer, without adding dynamic warts to the config?

If this field is present in the config on file, but the process has already finished and possibly another one has taken its PID, this can get messy.

**Review by mzabaluev — CHANGES_REQUESTED** [review]
Thanks for making it work.
I only have an objection about entangling the settlement client and the config with the vagaries of running an Anvil client process.

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (mzabaluev) [inline protocol-units/settlement/mcr/client/src/eth_client.rs:75]: 1. This a very happy-path solution for minding a child process. Let's leave it to process-compose or whatever other systemic solution we come up with.
2. There are in-process ways to kill a process, no need to run a command.
3. This is also not portable to non-Unix systems.

File: networks/suzuka/suzuka-full-node/src/main.rs
Comment (l-monninger) [reply networks/suzuka/suzuka-full-node/src/main.rs:43]: Since you won't be worry about merging on main until after this sprint, maybe we do want to spend a bit more time on style though.

File: protocol-units/settlement/mcr/config/src/lib.rs
Comment (mzabaluev) [reply protocol-units/settlement/mcr/config/src/lib.rs:28]: Filed this concern as #143 to unblock this PR.

File: networks/suzuka/suzuka-full-node/src/main.rs
Comment (musitdev) [reply networks/suzuka/suzuka-full-node/src/main.rs]: Yes removed, it was for test.

File: networks/suzuka/suzuka-full-node/src/main.rs
Comment (musitdev) [reply networks/suzuka/suzuka-full-node/src/main.rs:58]: yes I remove it.

File: protocol-units/da/m1/setup/src/local.rs
Comment (musitdev) [reply protocol-units/da/m1/setup/src/local.rs]: I'm used to remove all blank line in use statement because rustfmt sort them automatically if there are all in the same block. Sorted use is very convenient when you search for one. But I can change and keep your separate block.

File: protocol-units/settlement/mcr/client/src/eth/mod.rs
Comment (musitdev) [reply protocol-units/settlement/mcr/client/src/eth/mod.rs:3]: yes remove it.

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (musitdev) [reply protocol-units/settlement/mcr/client/src/eth_client.rs:30]: I don't agree here. They are sorted and you can find one easily. If you separate them like in this example. Alloy is separated. Automatic import when they are missing (my IDE add them automatically) will be added to the alloy section which result a real mash up where you have two sorted collections of import and need to search in both. And if you add more separate collection, in the end everything is mashed. Otherwise you've to very each time you import that the new import is in the right place that remove all the interest of automatic import.

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (musitdev) [reply protocol-units/settlement/mcr/client/src/eth_client.rs]: I'm agree. I do it like that because, in the std lib I didn't find a way to do it and I didn't want to import a new crate because it's a temporary solution until we settlement the config and process management. So it's working for mac and linux that are the current OS that we target that why I use this very easy solution.

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (musitdev) [reply protocol-units/settlement/mcr/client/src/eth_client.rs:85]: Yes it's a temporary solution until process management is settlement. Currently I've no other way to indicate a process from one process to the other.

File: protocol-units/settlement/mcr/config/src/lib.rs
Comment (musitdev) [reply protocol-units/settlement/mcr/config/src/lib.rs:28]: It's for local testing in a managed env. If anvil stop it won't be restarted unless the user explicitly do it.

File: util/commander/src/lib.rs
Comment (musitdev) [reply util/commander/src/lib.rs:85]: Yes good catch.

File: protocol-units/da/m1/setup/src/local.rs
Comment (mzabaluev) [reply protocol-units/da/m1/setup/src/local.rs]: Yes, sorting within groups is fine, but I like to keep the imports grouped by provenance as well.

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (mzabaluev) [reply protocol-units/settlement/mcr/client/src/eth_client.rs:30]: OK, this seems less of a headache with automatically added imports.

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (mzabaluev) [reply protocol-units/settlement/mcr/client/src/eth_client.rs:284]: To be further addressed in #105.

**Review by mzabaluev — APPROVED** [review]
Looks good to go, all concerns have been discussed and filed away to be fixed later if necessary.

File: networks/suzuka/suzuka-config/src/lib.rs
Comment (l-monninger) [inline networks/suzuka/suzuka-config/src/lib.rs:1]: This will always be used within Suzuka, so I think not flattening as you've done is fine.

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (l-monninger) [reply protocol-units/settlement/mcr/client/src/eth_client.rs:30]: I also prefer to group imports a bit.

File: protocol-units/settlement/mcr/client/src/eth_client.rs
Comment (l-monninger) [inline protocol-units/settlement/mcr/client/src/eth_client.rs:1]: There are still typos here. Please install the Grammarly extension and check the blue squiggles.

File: protocol-units/settlement/mcr/setup/src/local.rs
Comment (l-monninger) [inline protocol-units/settlement/mcr/setup/src/local.rs:1]: Looks reasonable enough.

File: networks/suzuka/setup/src/local.rs
Comment (l-monninger) [inline networks/suzuka/setup/src/local.rs:35]: Yep, looks good.

File: util/commander/src/lib.rs
Comment (l-monninger) [inline util/commander/src/lib.rs:82]: Yeah, reasonable way to do this.

**Review by l-monninger — CHANGES_REQUESTED** [review]
Please fix the typos, then we can merge.

**Review by l-monninger — APPROVED** [review]

**Comment by mzabaluev** [comment]
What causes the setup failure here: https://github.com/movementlabsxyz/movement/actions/runs/9775222670/job/26986389472?pr=139

**Pull request merged**
